docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+TODO Document how we are going to replace severity (currently prototyping)
+
+### Changes to Machine resource
+
+#### Machine Status
+
+Following changes are implemented to Machine's status:
+
+- Disambiguate usage of ready term by renaming fields used for the provisioning workflow
+- Align to K8s API conventions by deprecating `Phase` and corresponding `LastUpdated`
+- Remove `FailureReason` and `FailureMessage` to get rid of the confusing concept of terminal failures
+- Transition to new, improved, K8s API conventions aligned conditions
+
+Below you can find the relevant fields in Machine Status v1beta2, after v1beta1 removal (end state);
+After golang types, you can find a summary table that also shows how changes will be rolled out according to K8s deprecation rules.
+
+```golang
+type MachineStatus struct {
+
+    // Initialization provides observations of the Machine initialization process.
+    // NOTE: fields in this struct are part of the Cluster API contract and are used to orchestrate initial Machine provisioning.
+    // The value of those fields is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's BootstrapSecret.
+    // +optional
+    Initialization *MachineInitializationStatus `json:"initialization,omitempty"`
+    
+    // Represents the observations of a Machine's current state.
+    // +listType=map
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+    
+    // Other fields...
+    // NOTE: `Phase`, `LastUpdated`, `FailureReason`, `FailureMessage` fields won't be there anymore
+}
+
+// MachineInitializationStatus provides observations of the Machine initialization process.
+type MachineInitializationStatus struct {
+
+    // BootstrapSecretCreated is true when the bootstrap provider reports that the Machine's boostrap secret is created.
+    // NOTE: this field is part of the Cluster API contract, and it is used to orchestrate initial Machine provisioning.
+    // The value of this field is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's BootstrapSecret.
+    // +optional
+    BootstrapSecretCreated bool `json:"bootstrapSecretCreated"`
+    
+    // InfrastructureProvisioned is true when the infrastructure provider reports that the Machine's infrastructure is fully provisioned.
+    // NOTE: this field is part of the Cluster API contract, and it is used to orchestrate initial Machine  provisioning.
+    // The value of this field is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's infrastructure.
+    // +optional
+    InfrastructureProvisioned bool `json:"infrastructureProvisioned"`
+}
+```
+
+| v1beta1 (current)             | v1beta2 (tentative Q1 2025)                              | v1beta1 removal (tentative Q1 2026)        |
```

Comment:
Is the third column actually v1beta3?

Reply:
based on  https://kubernetes.io/docs/reference/using-api/deprecation-policy/#fields-of-rest-resources, it will be possible to drop back compatibily columns when we stop serving v1beta1, but please double check my assumption.

Reply:
While it also was initially surprising to me the language there seems to be pretty clear
> When v1 is eventually removed, the deprecated field can be removed from v2.

(and the example is even with GA apiVersions not even just beta)

Reply:
Resolving, it seems that there is agreement on how to read the API guidelines